testing/test_multicall.py

```diff
@@ -197,3 +197,40 @@ def m2():
     with pytest.raises(exc):
         MC([m2, m1], {}).execute()
     assert l == ["m1 init", "m1 finish"]
+
+
+@hookimpl(hookwrapper=True)
+def m1(arg1, arg2, arg3):
+    yield
+
+
+@hookimpl
+def m2(arg1, arg2, arg3):
+    return arg1, arg2, arg3
+
+
+@hookimpl(hookwrapper=True)
+def w1(arg1, arg2, arg3):
+    yield
+
+
+@hookimpl(hookwrapper=True)
+def w2(arg1, arg2, arg3):
+    yield
+
+
+def inner_exec(methods):
+    return MC(methods, {'arg1': 1, 'arg2': 2, 'arg3': 3}).execute()
+
+@pytest.mark.benchmark
+def test_hookimpls_speed(benchmark):
+    benchmark(inner_exec, [m1, m2])
+
+
+@pytest.mark.benchmark
+def test_hookwrapper_speed(benchmark):
+    benchmark(inner_exec, [w1, w2])
```

Comment:
You could have something like 
```
execute = MC(methods, {'arg1': 1, 'arg2': 2, 'arg3': 3}).execute
benchmark(execute, [w1, w2])
```
to avoid including the wrapper overhead in your timings. This would matter if the execute method is really really fast.